src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/RespondToClaimSpecCallbackHandler.java

```diff
@@ -352,12 +357,17 @@ private CallbackResponse validateCorrespondenceApplicantAddress(CallbackParams c
     private CallbackResponse populateRespondent1Copy(CallbackParams callbackParams) {
         var caseData = callbackParams.getCaseData();
         var updatedCaseData = caseData.toBuilder()
-            .respondent1Copy(caseData.getRespondent1())
-            .respondent2Copy(caseData.getRespondent2())
-            .build();
+            .respondent1Copy(caseData.getRespondent1());
+
+        updatedCaseData.respondent1DetailsForClaimDetailsTab(caseData.getRespondent1());
+
+        if (ofNullable(caseData.getRespondent2()).isPresent()) {
```

Comment:
can be replaced with ofNullable(caseData.getRespondent2()).ifPresent(r2 -> {
  updatedCaseData.respondent2Copy(r2)
.respondent2DetailsForClaimDetailsTab(r2)
});